virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -150,23 +162,124 @@ def setup(self) -> None:
             ).T,
             dims=["layers", "cell_id"],
             coords={
-                "layers": np.arange(15),
+                "layers": np.arange(len(self.layer_roles)),
                 "layer_roles": ("layers", self.layer_roles),
                 "cell_id": self.data.grid.cell_id,
             },
             name="soil_moisture",
         )
 
+        # create initial air temperature profile with reference temperature at surface
+        # for first soil evaporation update.
+        self.data["air_temperature"] = (
+            xr.concat(
+                [
+                    DataArray(
+                        np.full(
+                            (
+                                len(self.layer_roles)
+                                - self.layer_roles.count("soil")
+                                - 2,
+                                len(self.data.grid.cell_id),
+                            ),
+                            np.nan,
+                        ),
+                        dims=["layers", "cell_id"],
+                    ),
+                    self.data["air_temperature_ref"]
+                    .isel(time_index=0)
+                    .expand_dims("layers"),
+                    DataArray(
+                        np.full(
+                            (
+                                self.layer_roles.count("soil") + 1,
+                                len(self.data.grid.cell_id),
+                            ),
+                            np.nan,
+                        ),
+                        dims=["layers", "cell_id"],
+                    ),
+                ],
+                dim="layers",
+            )
+            .assign_coords(
+                coords={
+                    "layers": np.arange(len(self.layer_roles)),
+                    "layer_roles": ("layers", self.layer_roles),
+                    "cell_id": self.data.grid.cell_id,
+                },
+            )
+            .rename("air_temperature")
+        )
+
+        # create initial relative humidity profile with reference humidity at surface
+        # for first soil evaporation update.
+        self.data["relative_humidity"] = (
+            xr.concat(
+                [
+                    DataArray(
+                        np.full(
+                            (
+                                len(self.layer_roles)
```

Comment:
I wonder if it would be better to calculate some of these things as intermediate variables so that this assignment step gets simpler. I feel like it would probably be a bit more readable